### PR TITLE
Add multi-threading support to both demo apps.

### PR DESCRIPTION
**Pull request by rogerta**
In the demo agent, create a thread pool that better simulates how real agents will like work.

In the demo client, use multiple threads to send requests.  This is enabled with the --threaded comment line arg.

### CODE REVIEW

File: demo/agent.cc [hunk 1]
```diff
@@ -83,7 +83,7 @@ int main(int argc, char* argv[]) {
   }
 
   auto handler = use_queue
-      ? std::make_unique<QueuingHandler>(delay, save_print_data_path)
+      ? std::make_unique<QueuingHandler>(8u, delay, save_print_data_path)
```
Comment (dominiquefc) [inline demo/agent.cc:86]: Won't this remove the flag support for custom delay?

File: demo/client.cc [hunk 2]
```diff
@@ -260,53 +280,63 @@ ContentAnalysisAcknowledgement BuildAcknowledgement(
...
   int err = client->Send(request, &response);
   if (err != 0) {
-    std::cout << "[Demo] Error sending request " << position << std::endl;
-    return 1;
+    aout.stream() << "[Demo] Error sending request " << request.request_token()
+           << std::endl;
   } else if (response.results_size() == 0) {
-    std::cout << "[Demo] Response " << position << " is missing a result"
-              << std::endl;
-    return 1;
```
Comment (dominiquefc) [inline demo/client.cc:275]: Since this doesn't return 1 anymore, shouldn't it `err = 1;` ?

File: demo/handler.h [hunk 3]
```diff
@@ -66,51 +66,53 @@ class Handler : public content_analysis::sdk::AgentEventHandler {
...
+      stream << "  Verdict: allow" << std::endl;
     }
 
-    std::cout << std::endl;
+    stream << std::endl;
 
     // If a delay is specified, wait that much.
     if (delay_ > 0) {
-      std::cout << "[Demo] delaying request processing for " << delay_ << "s" << std::endl;
+      //stream << "[Demo] delaying request processing for " << delay_ << "s" << std::endl;
```
Comment (nancy-lan-xiao) [inline demo/handler.h:88]: Is commenting out this line intentional?

File: demo/agent.cc (hunk 1 above)
Comment (nancy-lan-xiao) [reply demo/agent.cc:86]: How come? We can still set the same custom delay for all the threads right? Or you mean setting different delays

Comment (rogerta) [reply demo/agent.cc:86]: I don't think so.  Could add a new flag to support the number of threads to use in the pool.

File: demo/handler.h (hunk 3 above)
Comment (rogerta) [reply demo/handler.h:88]: Good catch!  I meant to move this to `ProcessRequests()`.  Done.

File: demo/agent.cc (hunk 1 above)
Comment (rogerta) [reply demo/agent.cc:86]: I added new flag for num of threads since Nancy also asked about that.

**Review by nancy-lan-xiao — APPROVED** [review]
thanks, lgtm

**Review by dominiquefc — APPROVED** [review]

**Pull request merged**